This patch undoes parts of the following upstream patches:

commit 19442df2cd9a5bc8ec4deded7046ea7aca1d50a2
Author: Kota Tsuyuzaki <tsuyuzaki.kota@lab.ntt.co.jp>
Date:   Mon Feb 20 14:12:43 2017 -0800

    Fix warning on automake tool

    ..........

    Change-Id: I0e79bed7755a1f286b746a70fcf56fdc972bfd5d

commit b046ff7091e710ecafb31b13678e0092fd1d3fd7
Author: Ondřej Nový <novy@ondrej.org>
Date:   Sun Sep 30 12:23:22 2018 +0200

    Fix building with LD_LIBRARY_PATH set

    ..........

    Change-Id: Ib7a1d6b839d4a207ee0471b55233e1ce5d958705

diff -ur liberasurecode-1.6.1/Makefile.am liberasurecode-1.6.1-p3/Makefile.am
--- liberasurecode-1.6.1/Makefile.am	2019-05-24 08:10:22.000000000 -0500
+++ liberasurecode-1.6.1-p3/Makefile.am	2020-02-05 21:12:35.031620770 -0600
@@ -46,20 +46,38 @@
 	   $(LN_S) liberasurecode/config_liberasurecode.h)
 
 test: check
-	@./test/liberasurecode_test
-	@./test/alg_sig_test
-	@./test/test_xor_hd_code
-	@./test/libec_slap
+	$(eval SONAMES := $(shell find $(abs_top_builddir) -name '*.so'))
+	$(eval SODIRS := $(dir $(SONAMES)))
+	$(eval LD_LIBRARY_PATH := LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_LIBRARY_PATH := DYLD_LIBRARY_PATH="$(DYLD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_FALLBACK_LIBRARY_PATH := DYLD_FALLBACK_LIBRARY_PATH=$(DYLD_FALLBACK_LIBRARY_PATH):"$(subst / ,/:,$(SODIRS))")
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		./test/liberasurecode_test
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		./test/alg_sig_test
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		./test/test_xor_hd_code
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		./test/libec_slap
  
 VALGRIND_EXEC_COMMAND = $(LIBTOOL_COMMAND) valgrind --tool=memcheck \
 	--error-exitcode=1 --leak-check=yes --track-fds=yes \
 	--malloc-fill=A5 --free-fill=DE --fullpath-after=. --trace-children=yes
 
 valgrind-test: check
-	@$(VALGRIND_EXEC_COMMAND) ./test/alg_sig_test
-	@$(VALGRIND_EXEC_COMMAND) ./test/liberasurecode_test
-	@$(VALGRIND_EXEC_COMMAND) ./test/test_xor_hd_code
-	@$(VALGRIND_EXEC_COMMAND) ./test/libec_slap
+	$(eval SONAMES := $(shell find $(abs_top_builddir) -name '*.so'))
+	$(eval SODIRS := $(dir $(SONAMES)))
+	$(eval LD_LIBRARY_PATH := LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_LIBRARY_PATH := DYLD_LIBRARY_PATH="$(DYLD_LIBRARY_PATH):$(subst / ,/:,$(SODIRS))")
+	$(eval DYLD_FALLBACK_LIBRARY_PATH := DYLD_FALLBACK_LIBRARY_PATH=$(DYLD_FALLBACK_LIBRARY_PATH):"$(subst / ,/:,$(SODIRS))")
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		$(VALGRIND_EXEC_COMMAND) ./test/alg_sig_test
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		$(VALGRIND_EXEC_COMMAND) ./test/liberasurecode_test
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		$(VALGRIND_EXEC_COMMAND) ./test/test_xor_hd_code
+	@$(LD_LIBRARY_PATH) $(DYLD_LIBRARY_PATH) $(DYLD_FALLBACK_LIBRARY_PATH) \
+		$(VALGRIND_EXEC_COMMAND) ./test/libec_slap
 
 CLEANFILES = cscope.in.out cscope.out cscope.po.out
 
